btc/batcher.go

```diff
@@ -0,0 +1,524 @@
+package btc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/waddrmgr"
+	"github.com/btcsuite/btcwallet/wallet/txsizes"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"go.uber.org/zap"
+)
+
+var (
+	SegwitSpendWeight int = txsizes.RedeemP2WPKHInputWitnessWeight
+)
+var (
+	ErrBatchNotFound             = errors.New("batch not found")
+	ErrBatcherStillRunning       = errors.New("batcher is still running")
+	ErrBatcherNotRunning         = errors.New("batcher is not running")
+	ErrBatchParametersNotMet     = errors.New("batch parameters not met")
```

Comment:
The error is generic and does not specify what exactly went wrong. Maybe you can break the error into multiple errors and use them.